src/validation.cpp

```diff
@@ -211,14 +211,75 @@ bool CheckFinalTx(const CBlockIndex* active_chain_tip, const CTransaction &tx, i
     return IsFinalTx(tx, nBlockHeight, nBlockTime);
 }
 
+bool CalculateLockPoints(
+    CBlockIndex* tip,
+    const CCoinsView& coins_view,
+    const CTransaction& tx,
+    int flags,
+    LockPoints* lp)
+{
+    assert(tip);
+    assert(lp);
+
+    CBlockIndex index;
+    index.pprev = tip;
+    // CheckSequenceLocks() uses active_chainstate.m_chain.Height()+1 to evaluate
+    // height based locks because when SequenceLocks() is called within
+    // ConnectBlock(), the height of the block *being*
+    // evaluated is what is used.
+    // Thus if we want to know if a transaction can be part of the
+    // *next* block, we need to use one more than active_chainstate.m_chain.Height()
+    index.nHeight = tip->nHeight + 1;
+
+    std::vector<int> prevheights;
+    prevheights.resize(tx.vin.size());
+    for (size_t txinIndex = 0; txinIndex < tx.vin.size(); ++txinIndex) {
+        const CTxIn& txin = tx.vin[txinIndex];
+        Coin coin;
+        if (!coins_view.GetCoin(txin.prevout, coin)) {
+            return error("%s: Missing input", __func__);
```

Comment:
Can be replaced with a `LogPrintf` and `return std::nullopt`. Also seems like this log could be improved by adding the transaction id and/or prevout that is missing.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/23897#issuecomment-1102831191).